I2610 - dettl can use vault secrets for connecting to DB

r-ash opened the pull request:
* Copied code (and test) from orderly which manages reading vault secrets
* Added tests to check that we can use these to access a remote DB using creds from vault
* Added notes to README pointing to setup information about using vault

richfitz commented on R/vault.R:
```diff
@@ -0,0 +1,33 @@
+#'
+#' @keywords internal
+#'
+resolve_secrets <- function(x, config) {
```
If we're using this in both `orderly` and in `dettl`, do you think it should go into `vaultr` perhaps?

r-ash replied:
I think that would be a good idea. It it identical between `orderly` and `dettl`.

richfitz commented on R/dettl_config.R:
```diff
@@ -46,7 +46,7 @@ dettl_locate_config <- function(path) {
 db_config_read_yaml <- function(filename, path) {
   info <- yaml_read(filename)
-  check_fields(info, filename, "source", "destination")
+  check_fields(info, filename, "source", c("destination", "vault_server"))
```
It is probably worth validating the type of the `vault_server` field when present

richfitz commented on tests/testthat/test-db.R:
```diff
@@ -1,12 +1,30 @@
+  dir.create(path)
+  filename <- file.path(path, "db_config.yml")
+  cfg <- readLines("uat_config/db_config.yml")
+  cfg_server <- gsub(pattern = "<vault_server>", replace = srv$addr, x = cfg)
```
Tiny thing, but I would be inclined to used `fixed = TRUE` here also

r-ash commented:
Okay I think is ready for another look. I've addressed the points raised including moving the work to resolve secrets into `vaultr`. I've also expanded the test coverage (which is now working again after the tests have been rewritten) .
